File: android/src/main/java/com/lightningkite/khrysalis/observables/binding/SeekBar.binding.actual.kt
package com.lightningkite.khrysalis.observables.binding

import android.widget.SeekBar
import com.lightningkite.khrysalis.observables.*
import com.lightningkite.khrysalis.rx.removed
import com.lightningkite.khrysalis.rx.until

fun SeekBar.bind(
    start: Int,
    endInclusive: Int,
    observable: MutableObservableProperty<Int>
) {
    this.max = endInclusive - start
    this.incrementProgressBy(1)

    var suppress = false
    observable.subscribeBy { value ->
        if (!suppress) {
            suppress = true
            this.progress = value + start
            suppress = false
        }
    }.until(this@bind.removed)
    this.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
        override fun onProgressChanged(p0: SeekBar?, p1: Int, p2: Boolean) {
            if (!suppress) {
                suppress = true
                observable.value = p1
                suppress = false
            }
        }

        override fun onStartTrackingTouch(p0: SeekBar?) {
        }

        override fun onStopTrackingTouch(p0: SeekBar?) {
        }
    })

}

File: ios/Khrysalis/com/lightningkite/khrysalis/observables/binding/SeekBar.binding.actual.swift
//Stub file made with Khrysalis 2 (by Lightning Kite)
import Foundation
import UIKit


//--- SeekBar.bind(Int, Int, MutableObservableProperty<Int>)
public extension UISlider {
    func bind(_ start: Int32, _ endInclusive: Int32, _ observable: MutableObservableProperty<Int32>) -> Void {
        var suppress = false
        self.minimumValue = Float(start)
        self.maximumValue = Float(endInclusive)
        self.addAction(for: .valueChanged, action: { [weak self] in
            guard let self = self, !suppress else { return }
            suppress = true
            observable.value = Int32(self.value.rounded())
            suppress = false
        })
        observable.subscribeBy { (value) in
            guard !suppress else { return }
            suppress = true
            self.setValue(Float(value), animated: false)
            suppress = false
        }.until(self.removed)
    }
    func bind(start: Int32, endInclusive: Int32, observable: MutableObservableProperty<Int32>) -> Void {
        return bind(start, endInclusive, observable)
    }
}
